src/client/tile.h

```diff
@@ -305,6 +306,9 @@ struct TileSpec
 	 * Returns true if this tile can be merged with the other tile.
 	 */
 	bool isTileable(const TileSpec &other) const {
+		if (g_settings->getBool("enable_dynamic_shadows") &&
+			g_settings->getBool("shadow_psm"))
+				return false;
```

Comment:
This needs to be moved to the call site(s) or solved differently, this function should not call into g_settings every invocation.

Reply:
done.